### ISSUE TITLE
@SpringQueryMap doesn't work with inheritance

### ISSUE THREAD

**Issue by StasKolodyuk** [spring-cloud/spring-cloud-openfeign#168]
Hi,

@SpringQueryMap doesn't work properly when using inheritance. Suppose we have the following classes:
```
public class SimpleSearch {

    private String parentFilter;

    public String getParentFilter() {
        return parentFilter;
    }

    public void setParentFilter(String parentFilter) {
        this.parentFilter = parentFilter;
    }
}

public class ExtendedSearch extends SimpleSearch {

    private String childFilter;

    public String getChildFilter() {
        return childFilter;
    }

    public void setChildFilter(String childFilter) {
        this.childFilter = childFilter;
    }
}

@FeignClient(name = "postman", url = "https://postman-echo.com")
public interface PostmanEchoClient {

    @GetMapping("/get")
    void echo(@SpringQueryMap ExtendedSearch search);

}
```

When passing both *parentFilter* and *childFilter* only *childFilter* will apper on request:
```
    @Test
    public void test() {
        ExtendedSearch search = new ExtendedSearch();
        search.setParentFilter("parent");
        search.setChildFilter("child");

        // Request generated: https://postman-echo.com/get?childFilter=child
        // Expected:        : https://postman-echo.com/get?childFilter=child&parentFilter=parent
        postmanEchoClient.echo(search);
    }
```

Here's the [link](https://github.com/StasKolodyuk/spring-cloud-openfeign-query-map-issue) to the repo reproducing the issue. Have a look at `SpringQueryMapIssueTest` test.

Thanks

**Comment by FatTigerWang** [comment]
https://github.com/spring-cloud/spring-cloud-openfeign/issues/141

**Comment by spencergibb** [comment]
Closing in favor of https://github.com/OpenFeign/feign/issues/927

**Comment by kdavisk6** [comment]
https://github.com/OpenFeign/feign/pull/960.  Support will be available once Feign 10.3.0 is released.
